docs/dsl/middleware.md

```diff
@@ -285,3 +285,37 @@ object Example extends ZIOAppDefault {
 - [CORS](https://zio.github.io/zio-http/docs/v1.x/examples/advanced-examples/middleware_cors)
 - [CSRF](https://zio.github.io/zio-http/docs/v1.x/examples/advanced-examples/middleware_csrf)
 
+## Handler Aspects
+
+Ordinary Middlewares are intended to bracket a request's execution by intercepting the request, possibly modifying it or short-circuiting its execution
+(for example if it fails authentication), and then performing some post-processing on the response.
+However, we sometimes want to gather some contextual information about a request and pass it alongside to the request's handler.
+This can be achieved with the `HandlerAspect[Env, CtxOut]` type, which extends `Middleware[Env]`.
+This middleware produces a value of type `CtxOut` on each request, which the routing DSL will accept just like a path component:
+
+```scala
```

Comment:
Please try to use the `mdoc` as much as possible for snippet codes.

Reply:
Done, though I don't know how much that adds here, since these snippets won't compile as written.

Reply:
The goal is to make the code snippets maintainable by ensuring they can be compiled against the source code. So the "crash" modifier isn't helpful in this case.

Reply:
I didn't use `crash` but rather `fail` (the former is for code that compiles but fails at runtime, the latter for code that won't compile in the first place).

In any case, I had to resort to the hack of a `mdoc:invisible` block to declare some names.